lib/src/components/NxTextInput/NxTextInput.tsx

```diff
@@ -58,8 +58,8 @@ const NxTextInput = forwardRef<TextInputElement, Props>(
           firstValidationError = validatable && getFirstValidationError(validationErrors),
           internalClassName = classnames('nx-text-input', className, {
             pristine: isPristine,
-            invalid: validatable && isInvalid,
-            valid: validatable && !isInvalid,
+            invalid: !isPristine && validatable && isInvalid,
+            valid: !isPristine && validatable && !isInvalid,
```

Comment:
In an `NxForm`, you'll typically want to compute validation errors even for pristine inputs, as they should affect the disabled state of the Submit button (e.g. if there are required fields the button should initially be disabled until they are filled in).  The changes here and the `stateHelpers` make that much more convenient.